test/templates/client_test.rb

```diff
@@ -0,0 +1,67 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'test_helper'
+
+class HelloTest < Minitest::Test
+  def test_client
+    provider = ActionController::Base.new()
+    provider.prepend_view_path("templates")
+    service = OpenStruct.new(name: "Speech")
+    assigns = {
+      service: service,
+      namespaces: %w[google cloud speech v1]
+    }
+    result = provider.render_to_string(
+      template: "client",
+      layout: "ruby",
+      assigns: assigns
+    )
+    assert_equal expected, result
+  end
+
+  ##
+  # In the actual generator, the Ruby code in this output is formatted by
+  # Rubocop after it is rendered.
+  def expected
```

Comment:
Do you think this approach will scale? I think it's fine for now, but I wonder if it'll get awkward as the files get longer and the formatting is odd?

Reply:
I agree this approach fine for now, and also agree that comparing the full output will not scale as the templates change over time.

Reply:
Yes, this should be considered temporary. But at least the project now has a test! ;)

Reply:
What should we call the directory that holds the expected output files for generator tests?

Reply:
Something like `test/expected_output`, I think.